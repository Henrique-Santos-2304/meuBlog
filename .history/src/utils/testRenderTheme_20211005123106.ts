import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
