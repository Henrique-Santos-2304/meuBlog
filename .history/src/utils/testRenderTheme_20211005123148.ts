import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactElement): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
