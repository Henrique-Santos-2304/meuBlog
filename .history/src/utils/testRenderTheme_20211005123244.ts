import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
