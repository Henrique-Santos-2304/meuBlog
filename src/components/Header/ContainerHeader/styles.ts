import styled, { css } from "styled-components";
import { MdOutlineDragIndicator } from "react-icons/md";

export const Wrapper = styled.header`
  min-width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const Logo = styled.div`
  padding: 0 2rem;
`;

export const PreText = styled.h1`
  ${({ theme }) => css`
    font-family: ${theme.font.family.Tangerine};
    font-size: 4.5rem;
    color: ${theme.colors.mainLight};
    span {
      color: ${theme.colors.textTitles};
    }

    @media (min-width: ${theme.media.medium}) {
      font-size: 5.2rem;
    }
  `}
`;

export const ButtonMenu = styled.button`
  cursor: pointer;
  transition: all 0.4s linear;
  &:hover {
    filter: brightness(0.7);
  }
`;

export const IconMenu = styled(MdOutlineDragIndicator)`
  font-size: 4rem;
  color: ${({ theme }) => theme.colors.mainLight};
`;
