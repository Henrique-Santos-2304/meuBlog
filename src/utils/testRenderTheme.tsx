import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
