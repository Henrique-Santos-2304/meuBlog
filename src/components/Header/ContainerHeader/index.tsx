import * as S from "./styles";
import { useDisclosure } from "@chakra-ui/react";

import Menu from "components/globalComponents/Menu";
import Container from "components/globalComponents/Container";

const ContainerHeader = () => {
  const { isOpen, onOpen, onClose } = useDisclosure();
  return (
    <Container asHeight="medium">
      <S.Wrapper>
        <S.Logo aria-label="logo do site webdev">
          <S.PreText>
            Web<span>Dev</span>
          </S.PreText>
        </S.Logo>

        <S.ButtonMenu
          onClick={onOpen}
          aria-label="Botão para abrir um menu de opções"
          title="Abra o Menu"
        >
          <S.IconMenu />
        </S.ButtonMenu>

        <Menu onClose={onClose} isOpen={isOpen} />
      </S.Wrapper>
    </Container>
  );
};
export default ContainerHeader;
