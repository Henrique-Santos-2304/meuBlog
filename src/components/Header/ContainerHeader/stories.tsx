import { Story, Meta } from "@storybook/react";
import ContainerHeader from ".";

export default {
  title: "ContainerHeader",
  component: ContainerHeader,
} as Meta;

export const Basic: Story = (args) => <ContainerHeader {...args} />;

Basic.args = {
  title: "title basic",
  description: "description basic",
};

export const Default: Story = (args) => <ContainerHeader {...args} />;
