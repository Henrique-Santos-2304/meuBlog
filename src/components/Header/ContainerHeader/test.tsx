import ContainerHeader from ".";
import { screen, fireEvent } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";
import "jest-styled-components";

describe("<ContainerHeader />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<ContainerHeader />);
    expect(container).toMatchSnapshot();
  });
  it("should have childrens in the header ", () => {
    renderTheme(<ContainerHeader />);
    expect(screen.getByText(/web/i)).toBeInTheDocument();
    expect(screen.getByText(/dev/i)).toBeInTheDocument();
  });
  it("should open and close menu with click button ", () => {
    renderTheme(<ContainerHeader />);
    const buttonOpen = screen.getByRole("button", {
      name: /Botão para abrir um menu de opções/i,
    });

    expect(buttonOpen).toBeInTheDocument();

    fireEvent.click(buttonOpen);
    expect(
      screen.getByRole("button", { name: /Botão para fechar um menu/i })
    ).toBeInTheDocument();
  });
});
