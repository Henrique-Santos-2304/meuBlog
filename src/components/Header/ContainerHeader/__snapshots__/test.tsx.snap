// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContainerHeader /> should render  1`] = `
.c2 {
  min-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  padding: 0 2rem;
}

.c4 {
  font-family: 'Tangerine',cursive;
  font-size: 4.5rem;
  color: #f8f8f8;
}

.c4 span {
  color: #F4F66F;
}

.c5 {
  cursor: pointer;
  -webkit-transition: all 0.4s linear;
  transition: all 0.4s linear;
}

.c5:hover {
  -webkit-filter: brightness(0.7);
  filter: brightness(0.7);
}

.c6 {
  font-size: 4rem;
  color: #f8f8f8;
}

.c0 {
  width: 100vw;
  background-image: linear-gradient( to left,#00bf72,#008793,#004d7a,#051937 );
}

.c0 > div {
  min-height: 10vh;
}

.c1 {
  max-width: 86rem;
  padding: 2rem;
  margin: 0 auto;
}

@media (min-width:767px) {
  .c4 {
    font-size: 5.2rem;
  }
}

<div>
   
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <header
        class="c2"
      >
        <div
          aria-label="logo do site webdev"
          class="c3"
        >
          <h1
            class="c4"
          >
            Web
            <span>
              Dev
            </span>
          </h1>
        </div>
        <button
          aria-label="Botão para abrir um menu de opções"
          class="c5"
          title="Abra o Menu"
        >
          <svg
            class="c6"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </button>
      </header>
    </div>
  </div>
</div>
`;
